lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Replicator.java

```diff
@@ -252,29 +252,33 @@ public C4ReplicatorStatus getStatus() {
         }
     }
 
-    public Set<String> getPendingDocIDs() throws LiteCoreException {
+    @Nullable
+    public byte[] getResponseHeaders() {
         synchronized (lock) {
-            if (handle == 0) { return null; }
-
-            final FLSliceResult result = new FLSliceResult(getPendingDocIds(handle));
-            try {
-                final FLValue slice = FLValue.fromData(result);
-                return (slice == null) ? Collections.emptySet() : new HashSet<>(slice.asTypedArray());
-            }
-            finally { result.free(); }
+            return (handle == 0) ? null : getResponseHeaders(handle);
         }
     }
 
-    public boolean isDocumentPending(String docId) throws LiteCoreException {
+    // Null return value indicates that this replicator is dead
+    @Nullable
+    public Boolean isDocumentPending(String docId) throws LiteCoreException {
         synchronized (lock) {
-            return (handle != 0) && isDocumentPending(handle, docId);
+            return (handle == 0) ? null : Boolean.valueOf(isDocumentPending(handle, docId));
         }
     }
 
+    // Null return value indicates that this replicator is dead
     @Nullable
-    public byte[] getResponseHeaders() {
+    public Set<String> getPendingDocIDs() throws LiteCoreException {
         synchronized (lock) {
-            return (handle == 0) ? null : getResponseHeaders(handle);
+            if (handle == 0) { return null; }
+
+            final FLSliceResult result = new FLSliceResult(getPendingDocIds(handle));
```

Comment:
not sure, whether this is valid in case of Java, but `getPendingDocIds` is already returning a `FLSliceResult`, and we are using the same result to create another `FLSliceResult`.

Reply:
Whoah!  Lemme check that.  I think that is just wrapping in the FLSliceResult Java object, but I better check!  Thx!

Reply:
Yeah... its all good.  `getPendingDocIds` returns a native reference to a  native`FLSliceResult`.  `new FLSliceResult` wraps the native reference in a Java object.